### DISCUSSION TITLE
Loading state for chat messages

### DISCUSSION THREAD

**Discussion by tonge495** [twilio-labs/paste#3482] [Q&A] [answered]
Hi team,

I am working on CommsGPT, a project where we are creating a ChatGPT powered conversational UI platform on One Admin.  I was wondering are there any design patterns for the loading state for new chat messages?

![comms_gpt_convo_wall](https://github.com/twilio-labs/paste/assets/9394413/4c7c313b-2609-4742-b62e-a0ae254a067c)

Any help would be appreciated, thank you!

Eileen

**Comment by serifluous** [comment]
By loading state, do you mean something to indicate that someone's typing?

**Reply by tonge495** [reply]
Yes that's right!  So when CommsGPT is writing a message to the user. In the backend, sometimes it needs to think a bit so the response will not always be immediate.

**Reply by serifluous** [reply, marked as answer]
ok cool! We don't have it yet, but it is something we could add as part of the AI UI kit coming up.

**Reply by tonge495** [reply]
That would be great!  We are currently in the development stage though and would probably need to make an interim solution in the meantime.  Once the AI UI kit is up and running, we can always swap it.  Happy to also provide feedback fro the AI UI kit as well.

Would that work on your end?

**Reply by serifluous** [reply]
Yep, works perfectly

**Reply by serifluous** [reply]
By the way, have you tried using the Chat Composer instead of a Textarea to post messages? https://paste.twilio.design/components/chat-composer#with-chat-log

**Reply by tonge495** [reply]
I have not, but I can suggest it to the developers, thank you!

**Reply by tonge495** [reply]
Quick question, is there a reason why in the Figma component there is a top border for the chat composer whereas the in the Paste documentation, there is no line but a scrollbar?  Is there a specific reason why there is only a top border?

Also is there an inline error state for the chat composer?

Figma component:
<img width="603" alt="Screenshot 2023-09-11 at 5 09 47 PM" src="https://github.com/twilio-labs/paste/assets/9394413/4f2e7ceb-e0dd-4bfa-b4da-c1b8439b2e48">

Paste documentation:
![Screenshot 2023-09-11 at 5 10 32 PM](https://github.com/twilio-labs/paste/assets/9394413/34617956-b6f9-455e-9144-1418acab100e)

**Reply by serifluous** [reply]
1. There's a top border because it's meant to be placed with Chat Log like this: https://paste.twilio.design/components/chat-composer#with-chat-log We didn't imagine many designers would try to use it separately from the Chat Log, so it comes with Chat Composer in Figma.
2. For the scrollbar, we don't typically create scrollbars for the Figma components for easier maintenance, and users will have different scrollbar visibility settings on. You'll see the same thing with Code Block, for example.
3. What kind of inline error are you imagining for chat composer?

**Reply by tonge495** [reply]
Thanks Sarah! For #3, we were thinking of having an inline error if the user tries to click on the arrow on the right without entering any text in the field.

**Reply by serifluous** [reply]
Do you think you need an inline error for that case? fwiw the send button can also be hidden or disabled. You can probably follow the usual chat/messaging paradigms for this, unless you've user-tested otherwise.

**Reply by tonge495** [reply]
Got it, we can just disable it then.  For #1, just to clarify is the chat composer supposed to be inside or outside of the chat box?

#1 Inside
![CommsGPT - CommsGPT loading (1)](https://github.com/twilio-labs/paste/assets/9394413/febdcb76-3d82-4946-84fd-3bddd90138df)


#2 Outside
![image](https://github.com/twilio-labs/paste/assets/9394413/938ea457-81a6-4262-870a-6c5ff197501e)

**Reply by serifluous** [reply]
It was designed mainly to be placed inside, but if you need on the outside, let us know.

We still need to put together a docs page with more guidelines on using Chat Composer and Chat Log together, but you can see an example sandbox here: https://paste.twilio.design/blog/2022-12-06-announcing-the-conversations-ui-kit

**Reply by tonge495** [reply]
Thank you!  Sorry about the 1 million questions.  I'm working with the developers this sprint on this and we were trying to figure out all the details. :)

**Reply by serifluous** [reply]
No worries! That's what Discussions is here for.
